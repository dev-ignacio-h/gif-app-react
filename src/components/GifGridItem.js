const GifGridItem = ({id, title, url}) => {
    // console.log(img);
    // console.log(id)
    // console.log(title)
    // console.log(url)
    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    );
}
 
export default GifGridItem;