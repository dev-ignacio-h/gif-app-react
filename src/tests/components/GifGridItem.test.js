import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Tests in GifGridItem component', () => {
  const title = 'A title';
  const url = 'https://localhost/algo.jpg';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('should display the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have a paragraph with the title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('should have the image equal to the url and alt of the props', () => {
    const img = wrapper.find('img');
    // console.log(img.props().src === img.prop('src'));
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('should have class animate__bounce', () => {
    const div = wrapper.find('div')
    const className = div.prop('className')
    // console.log(div.prop('className').includes('animate__bounce'));
    expect(className.includes('animate__bounce')).toBe(true)
  });
});
