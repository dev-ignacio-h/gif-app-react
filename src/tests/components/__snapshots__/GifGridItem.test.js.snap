// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifGridItem component should display the component correctly 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="A title"
    src="https://localhost/algo.jpg"
  />
  <p>
    A title
  </p>
</div>
`;
